vmtest/enter_kdump.py

```diff
@@ -19,10 +20,15 @@
 with open("/proc/cmdline", "rb") as f:
     cmdline = f.read().rstrip(b"\n")
     cmdline = re.sub(rb"(^|\s)crashkernel=\S+", b"", cmdline)
-    # `nosmp` is required to avoid QEMU sporadically failing an internal assertion
     # `nokaslr` is required to avoid sporadically failing to reserve space for the
     # capture kernel
-    cmdline += b" nosmp nokaslr"
+    cmdline += b" nokaslr"
+    if (
```

Comment:
I tweaked this to only enable `nosmp` when not using KVM (since that seems to be the only time the assertion is triggered in QEMU) because not using KVM makes the tests run much slower.

Reply:
We're not guaranteed that `systemd-detect-virt` exists on the host system. Is there another way to detect this? Or maybe we can pass this information to the VM explicitly.

Reply:
Some googling didn't suggest much past `dmidecode`, `systemd-detect-virt`, and a variety of unreliable checks which require `grep`'ing through the output of `dmesg`. I don't know that it's possible to pass this info to the VM explicitly without sacrificing reliability, as I believe it's possible for us to _attempt_ using KVM acceleration (and thus specify that to the VM), but for it to fail and fall back to software emulation.

My thought here was that this should degrade gracefully – if `systemd-detect-virt` isn't found, then the output won't match and thus `nosmp` will be enabled, leaving us no worse off than the behavior without this tweak. In other words, it can only help.

Reply:
From my testing, if you pass `-enable-kvm` to QEMU, it won't fall back to not using KVM. (That's why `run_in_vm` explicitly checks that it can access `/dev/kvm` before calling QEMU.) So I think we can get away with passing something in. That also gives us the flexibility to avoid this workaround based on a version check if the bug is fixed in the future. (Side note, we should report this to QEMU upstream.)

Reply:
Also, this change could be its own PR, but at the very least it should be a separate commit in this PR.

Reply:
Extracted it into its own commit and modified it as you suggested.